src/OpenTelemetry.Api/Metrics/MeterFactoryBase.cs

```diff
@@ -32,14 +37,50 @@ public static MeterFactoryBase Default
         }
 
         /// <summary>
-        /// Returns an IMeter for a given name and version.
+        /// Sets the default instance of <see cref="MeterFactoryBase"/>.
+        /// </summary>
+        /// <param name="meterFactory">Instance of <see cref="MeterFactoryBase"/>.</param>
+        /// <remarks>
+        /// This method can only be called once. Calling it multiple times will throw an <see cref="System.InvalidOperationException"/>.
+        /// </remarks>
+        /// <exception cref="System.InvalidOperationException">Thrown when called multiple times.</exception>
+        public static void SetDefault(MeterFactoryBase meterFactory)
+        {
+            if (isInitialized)
+            {
+                throw new InvalidOperationException("Default factory is already set");
```

Comment:
TracerFactory is doing this same thing - I think we can make it NOT throw, and log an error.
Will be following up on this aspect.

Reply:
I think it's worth discussing. The question is whether we want to hide this error or want to crash an app to indicate that something is very broken if it is being set twice. Worth having an issue for the discussion

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/issues/638